### DISCUSSION TITLE
Best method to split openapi into multiple files

### DISCUSSION THREAD

**Discussion by Christoph-AK** [poem-web/poem#102] [Q&A] [answered]
Hey there,
I want to migrate a project with multiple endpoints to poem_openapi.

In actix I could define all my functions and structs in their own modules and call `server.get("/ep1", mod1::fn1).post("/ep2", mod2::fn1)` and so on. This makes for a nice, tidy main and structured modules.

If I understand correctly in poem every endpoint definition needs to be in the same file in the same impl block. 

Of cause I could treat every function as a wrapper, so for example 

    #[oai(path = "/test", method = "get")]
    async fn test(&self, auth: authmod::struct1) -> PlainText<mod1::struct2> {
        mod1::test(&self, auth)
    }

    #[oai(path = "/test2", method = "get")]
    async fn test2(&self, auth: authmod::struct1) -> PlainText<mod2::struct1> {
        mod2::test(&self, auth)
    }

but that is quite a handfull to maintain too.

Is there maybe a way to glue multiple API-definitions into one?

**Comment by sunli829** [comment, marked as answer]
use the `combine` method to merge multiple OpenAPIs into one.

https://github.com/poem-web/poem/blob/da103bee5fe79eaf6c0d20eac3d3b4b691e56137/examples/openapi/combined-apis/src/main.rs#L41

**Comment by Christoph-AK** [comment]
Thank you! I seem to be absolutely blind today.

**Comment by invokermain** [comment]
The answer is outdated, nowadays `OpenApiSerivce::new` can be passed a tuple to combine multiple API objects.
